test/autorange.jl

```diff
@@ -0,0 +1,114 @@
+@testset "CornerIterator" begin
+    # check that CornerIterator is not exported
+    @test_throws UndefVarError CornerIterator
+
+    # test interface and behaviour under normal circumstances
+    # (i.e. without any singleton array dimension of size 1)
+    si(num::Number) = 1
+    ei(num::Number) = num
+    si(num::Range)  = minimum(num)
+    ei(num::Range)  = maximum(num)
+    for t in ((2,), (2,3), (2,3,4), (2,3,4,5),        # OneTo
+              (-1:2,), (-1:2,-2:3), (-1:2,-2:3,-3:4)) # Non OneTo
+        N = length(t)
+        @testset "$N-d array without singleton dimension" begin
+            cr = CartesianRange(t)
+            ci = @inferred(ImageTransformations.CornerIterator(cr))
+            @test typeof(ci) <: ImageTransformations.CornerIterator{CartesianIndex{N}}
+            @test ci.start === cr.start
+            @test ci.stop === cr.stop
+            @test @inferred(eltype(ci)) === CartesianIndex{N}
+            @test @inferred(size(ci)) === ntuple(x->2, N)
+            @test @inferred(length(ci)) === 2^N
+            # Check for correct shape and values
+            A = @inferred(collect(ci))
+            @test typeof(A) <: Array{CartesianIndex{N},N}
+            if N == 1
+                @test A == [CartesianIndex{1}((si(t[1]),)), CartesianIndex{1}((ei(t[1]),))]
```

Comment:
Can be written more simply as `CartesianIndex(si(t[1]))`. Likewise below for `N > 1`.